### PR TITLE
Turning on HoistCommonInsts flag for CFGSimplificationPass.

### PR DESCRIPTION
**Pull request by DeepakRajendrakumaran**

### CODE REVIEW

File: src/opt.cpp [hunk 1]
```diff
@@ -482,6 +482,10 @@ void Optimize(llvm::Module *module, int optLevel) {
     }
     optPM.add(llvm::createIndVarSimplifyPass());
 
+#if ISPC_LLVM_VERSION > ISPC_LLVM_11_0
```
Comment (dbabokin) [inline src/opt.cpp:485]: This needs to be >= LLVM 12 to handle correctly LLVM 11.1

```suggestion
#if ISPC_LLVM_VERSION >= ISPC_LLVM_12_0
```

File: src/opt.cpp [hunk 2]
```diff
@@ -517,7 +521,11 @@ void Optimize(llvm::Module *module, int optLevel) {
 #endif
         optPM.add(llvm::createFunctionInliningPass());
         optPM.add(CreateMakeInternalFuncsStaticPass());
+#if ISPC_LLVM_VERSION <= ISPC_LLVM_11_0
```
Comment (dbabokin) [inline src/opt.cpp:524]: The same, LLVM 11.1 needs to be handled correctly.
So it needs to be ">= 12.0" and "else".

**Review by dbabokin — CHANGES_REQUESTED** [review]

File: src/opt.cpp (hunk 1 above)
Comment (DeepakRajendrakumaran) [reply src/opt.cpp]: Done

File: src/opt.cpp (hunk 2 above)
Comment (DeepakRajendrakumaran) [reply src/opt.cpp]: Done

**Review by dbabokin — APPROVED** [review]

**Comment by dbabokin** [comment]
LGTM

**Pull request merged**